games-roguelike/dwarf-fortress/files/dwarf-fortress-0.47.05-nogtk.patch

```diff
@@ -0,0 +1,147 @@
+Removes hard dependency on GTK in favour of stdin calls.
+
+diff --git a/g_src/KeybindingScreen.cpp b/g_src/KeybindingScreen.cpp
+index 5dd3f2f..4d1a108 100644
+--- a/g_src/KeybindingScreen.cpp
++++ b/g_src/KeybindingScreen.cpp
+@@ -1,7 +1,9 @@
+ #ifdef __APPLE__
+ # include "osx_messagebox.h"
+ #elif defined(unix)
+-# include <gtk/gtk.h>
++# ifdef HAVE_GTK
++#  include <gtk/gtk.h>
++# endif
+ #endif
+ 
+ #include "GL/glew.h"
+diff --git a/g_src/enabler.cpp b/g_src/enabler.cpp
+index 897d56a..4ea9458 100644
+--- a/g_src/enabler.cpp
++++ b/g_src/enabler.cpp
+@@ -1,7 +1,9 @@
+ #ifdef __APPLE__
+ # include "osx_messagebox.h"
+ #elif defined(unix)
+-# include <gtk/gtk.h>
++# ifdef HAVE_GTK
++#  include <gtk/gtk.h>
++# endif
+ #endif
+ 
+ #include <cassert>
+@@ -715,7 +717,7 @@ int main (int argc, char* argv[]) {
+ #ifdef unix
+   setlocale(LC_ALL, "");
+ #endif
+-#if !defined(__APPLE__) && defined(unix)
++#if !defined(__APPLE__) && defined(unix) && defined(HAVE_GTK)
+   bool gtk_ok = false;
+   if (getenv("DISPLAY"))
+     gtk_ok = gtk_init_check(&argc, &argv);
+@@ -736,6 +738,7 @@ int main (int argc, char* argv[]) {
+   init.begin(); // Load init.txt settings
+   
+ #if !defined(__APPLE__) && defined(unix)
++ #if defined(HAVE_GTK)
+   if (!gtk_ok && !init.display.flag.has_flag(INIT_DISPLAY_FLAG_TEXT)) {
+     puts("Display not found and PRINT_MODE not set to TEXT, aborting.");
+     exit(EXIT_FAILURE);
+@@ -745,6 +748,7 @@ int main (int argc, char* argv[]) {
+     puts("Graphical tiles are not compatible with text output, sorry");
+     exit(EXIT_FAILURE);
+   }
++ #endif
+ #endif
+ 
+   // Initialize video, if we /use/ video
+diff --git a/g_src/renderer_curses.cpp b/g_src/renderer_curses.cpp
+index d93061d..f873784 100644
+--- a/g_src/renderer_curses.cpp
++++ b/g_src/renderer_curses.cpp
+@@ -1,3 +1,7 @@
++#if defined(__APPLE__) || defined(unix)
++# include <unistd.h>
++#endif
++
+ static bool curses_initialized = false;
+ 
+ static void endwin_void() {
+diff --git a/g_src/ttf_manager.cpp b/g_src/ttf_manager.cpp
+index aeebc14..ce48a03 100644
+--- a/g_src/ttf_manager.cpp
++++ b/g_src/ttf_manager.cpp
+@@ -2,6 +2,7 @@
+ #include "init.h"
+ #include <cmath>
+ #include <iostream>
++#include <cmath>
```

Comment:
Is this really needed?

Reply:
Woops, I also used the patch to build on a stricter compiler and never removed it. Fixed in latest commit.